packages/expo-image/ios/ResizedAnimatedCoder.swift

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024-present 650 Industries. All rights reserved.
+
+import SDWebImage
+
+/**
+ An animated image resizer that adjusts an already loaded `SDAnimatedImage`.
+ This is not a formal SDWebImage coder; it's used solely to leverage the SDWebImage loading pipeline.
+ */
+internal final class ResizedAnimatedCoder: SDImageIOAnimatedCoder {
```

Comment:
Seems inherit from `SDImageIOAnimatedCoder` will not use the custom AVIF/WebP coder, this only supports Apple's System ImageIO format like JPEG/HEIC/WebP(from iOS 14+)

Reply:
the tricky part of the current solution is that even the `ResizedAnimatedCoder` inherits SDImageIOAnimatedCoder, it does not do any encode/decode. only do post-transform in `animatedImageFrameAtIndex`.

we used existing SDWebImage (which is decoded already), so it supports AVIF.

i was thinking your idea to store the `actualCoder`. i got blocked from the [eight protocols](https://github.com/SDWebImage/SDWebImage/blob/0b10fcb54484b618a53386670bff38176a203290/SDWebImage/Core/SDImageIOAnimatedCoder.h#L24-L56) are class static. it's difficult to pass from instance `actualCoder`.

Reply:
> i got blocked from the [eight protocols](https://github.com/SDWebImage/SDWebImage/blob/0b10fcb54484b618a53386670bff38176a203290/SDWebImage/Core/SDImageIOAnimatedCoder.h#L24-L56) are class static

No, The coder protocol APIs are all instance method.

Your link pointed to another ImageIO based coder implementation details, it's different thing and you don't need them at all. 

https://sdwebimage.github.io/documentation/sdwebimage/sdimagecoder

Reply:
oh interesting! let me try for SDImageCoder. thanks for the hint!